src/game/SpellEffects.cpp

```diff
@@ -3700,15 +3714,19 @@ void Spell::DoSummon(SpellEffectIndex eff_idx)
 
     if (m_caster->GetTypeId() == TYPEID_PLAYER)
     {
-        // this enables popup window (pet dismiss, cancel)
         spawnCreature->SetUInt32Value(UNIT_FIELD_FLAGS, UNIT_FLAG_PVP_ATTACKABLE);
 
-        spawnCreature->GetCharmInfo()->SetPetNumber(pet_number, true);
+        CharmInfo* charmInfo = spawnCreature->GetCharmInfo();
+        if (m_caster->getClass() == CLASS_PRIEST)
```

Comment:
The thing is that it is passive since it is not given a react state, and while defensive it would attack its target, on passive it just runs back to its owner, so we have to overwrite it to be either defensive or aggressive and aggressive makes more sense in my opinion. Are you sure the same does not go for DK gargoyles?